domain/entities/rule_engine.py

```
@@ -0,0 +1,184 @@
+            except Exception as e:
+                logger.error(f"Rule check failed: {rule_name}, error: {e}")
+                results[rule_name] = False
+        return results
+
+    def determine_stage_completion(self, stage_id: str, context: Dict) -> bool:
+        """确定指定阶段是否完成。
+
+        根据不同阶段的特定规则来判断阶段完成情况。
+
+        Args:
+            stage_id: 阶段 ID（如 stage_1, stage_2 等）。
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)